### ISSUE TITLE
[Bug] __LINE__ predeclared symbol does not work

### ISSUE THREAD

**Issue by ekimekim** [rednex/rgbds#206]
Steps to reproduce:

In file "example.asm":
```
PRINTT "{__FILE__}\n"
PRINTT "{__LINE__}\n"
```
Attempt to compile example.asm:
```
$ rgbasm example.asm
```

Expected result:

As per https://rednex.github.io/rgbds/rgbasm.5.html#__LINE__, the `__FILE__` and `__LINE__` symbols are expanded to the current filename and line number respectively, and printed:
```
"example.asm"
2
```

Actual result:

The filename prints correctly, but the `__LINE__` symbol does not exist:
```
"example.asm"
ERROR: example.asm(2):
	'__LINE__' not defined
$0
error: Assembly aborted in pass 1 (1 errors)!
```

Versions tested:

Current master (193cc06, v0.3.3), on linux.

Other notes:

I did a quick search of the codebase for `__FILE__` and `__LINE__`. In the case of `__FILE__` I find where it's defined: https://github.com/rednex/rgbds/blob/193cc06561775b9eaa5d63d19ffed8cdec9cb3d8/src/asm/fstack.c#L391 but for `__LINE__` I see no references apart from in the docs.

It looks like this feature was removed in 6198cc185c24ec4ba794ed27c969ae65d31f89c3 but the docs were never updated. In that case, please consider this issue both a bug that it was not removed from the docs, and a feature request to re-implement the feature.

As a final aside: I realise I've been demanding quite a lot in the last few days. I just want to say that I'm ever grateful for the work you all are doing on these tools, and I totally understand that you may be too busy to enact every request I make. Thank you for all your effort.

**Comment by AntonioND** [comment]
Interesting... After reverting the commit where it was removed, it kinda works. It prints the line number in the first pass of assembly, and in the second one it gives an error. I'll try to fix this.

Don't worry about asking for things. It's better to have a list of nice to haves than to not have anything. That way, if anyone wants to contribute, there's a list of ideas of what to do. :P

**Comment by AntonioND** [comment]
I've pushed this patch to the develop branch of the, here: https://github.com/rednex/rgbds/tree/develop

Commit https://github.com/rednex/rgbds/commit/90bc8d91100b23c6e4283c2983cc353434bb8956

Could you tell me if this works for you? As far as I can tell, it's working nicely in my PC.

**Comment by AntonioND** [comment]
Ok, turns out that it breaks Travis, I'll fix that warning tomorrow...

**Comment by ekimekim** [comment]
Yep, the `__LINE__` symbol seems to work now, though my attempts to use it in a debug message macro haven't gone so well - I was hoping it would act like `__FILE__` and return the line number where the macro was invoked in the main file. Instead, it seems to be returning the line number inside the macro. I can see why it would do that though.

**Comment by ekimekim** [comment]
Aha, I worked out how to do it. I need to pass the __LINE__ at the macro invocation site into the macro as an argument. To automate this, I use a `EQUS` with one layer of quoting so that the `__LINE__` is not resolved to a value immediately, but only once the EQUS is expanded. Like so:
```
Debug EQUS "_Debug \{__LINE__\}, "
_Debug: MACRO
  db strcat(__FILE__, strcat("@\1: ", \2))
ENDM
```
then use it like:
```
    Debug "foo bar baz"
```
which produces output like:
```
myfile.asm@$1A: foo bar baz
```
...so then, only thing left is to find some way to display this number in decimal instead.

**Comment by AntonioND** [comment]
I think that you should just create a macro to do this:

```
IF __LINE__ < 10
  DB '0' + __LINE__
ELIF __LINE__ < 100
  DB '0' + (__LINE__ / 10)
  DB '0' + (__LINE__ % 10)
ELIF ...
...
ENDC
```

The problem is in `src/asm/asmy.y:84`. I can't just change it to decimal because it might break someone else's code, and I don't see a simple way of fixing it. As you said, probably the best way to fix this is to implement something like https://github.com/rednex/rgbds/issues/178.

**Comment by AntonioND** [comment]
Fixed by having `PRINTI`, introduced here: https://github.com/rednex/rgbds/commit/d243bd04ef929aa99158496c6f5fb048ec55ffa1
